src/main/kotlin/org/kafkalytic/plugin/KafkaUtils.kt

```diff
@@ -223,7 +228,15 @@ fun produceGeneratedMessages(producer: KafkaProducer<ByteArray, ByteArray>, topi
                 }.toString()
             }
 
-            channel.send(producer.send(ProducerRecord<ByteArray, ByteArray>(topic, "message$current".toByteArray(), template.replace(RANDOM_PLACEHOLDER, value).toByteArray())))
+            val message = ProducerRecord<ByteArray, ByteArray>(topic, "message$current".toByteArray(), template.replace(RANDOM_PLACEHOLDER, value).toByteArray())
+            if(!header.isNullOrEmpty()){
+                //Custom WITHOUT HEADER
+                val header = createCustomHeader(header, current);
+                header.forEach {
+                    message.headers().add(it);
+                }
+            }
+            channel.send(producer.send(message));
```

Comment:
with new implementation of  createCustomHeader below it looks like:
            val message = ProducerRecord(topic, null,
                "message$current".toByteArray(),
                template.replace(RANDOM_PLACEHOLDER, value).toByteArray(),
                header?.let { createCustomHeader(it) }
            )